app/src/main/java/com/android/voyageur/ui/overview/Overview.kt

```diff
@@ -142,25 +168,50 @@ fun OverviewScreen(
-              TextField(
-                  value = searchQuery,
-                  onValueChange = { searchQuery = it },
-                  placeholder = {
-                    Text(
-                        text = stringResource(R.string.overview_searchbar_placeholder),
-                        style = MaterialTheme.typography.bodyLarge.copy(fontSize = 16.sp),
-                        modifier = Modifier.fillMaxWidth())
-                  },
-                  modifier =
-                      Modifier.height(56.dp) // Increased height
-                          .fillMaxWidth()
-                          .testTag("searchField"),
-                  textStyle = MaterialTheme.typography.bodyLarge.copy(fontSize = 16.sp),
-                  singleLine = true,
-                  shape = RoundedCornerShape(12.dp), // Slightly increased corner radius
-              )
+                    .padding(horizontal = 24.dp, vertical = 20.dp) // Increased padding
+            ) {
+              Row(
+                  modifier = Modifier.fillMaxWidth(),
+                  verticalAlignment = Alignment.CenterVertically) {
+                    // TextField on the left
+                    TextField(
+                        value = searchQuery,
+                        onValueChange = { searchQuery = it },
+                        placeholder = {
+                          Text(
+                              text = stringResource(R.string.overview_searchbar_placeholder),
+                              style = MaterialTheme.typography.bodyLarge.copy(fontSize = 16.sp),
+                              modifier = Modifier.fillMaxWidth())
+                        },
+                        modifier =
+                            Modifier.weight(
+                                    1f) // Forces TextField to take up all available horizontal
+                                // space
+                                .height(56.dp) // Increased height
+                                .testTag("searchField"),
+                        textStyle = MaterialTheme.typography.bodyLarge.copy(fontSize = 16.sp),
+                        singleLine = true,
+                        shape = RoundedCornerShape(12.dp) // Slightly increased corner radius
+                        )
+
+                    Spacer(modifier = Modifier.width(8.dp)) // Optional spacing
+
+                    // IconButton explicitly aligned to the right
+                    IconButton(
+                        onClick = { showOnlyFavorites = !showOnlyFavorites },
+                        modifier = Modifier.testTag("favoriteFilterButton")) {
+                          Icon(
+                              imageVector =
+                                  if (showOnlyFavorites) Icons.Filled.Favorite
+                                  else Icons.Default.FavoriteBorder,
+                              contentDescription =
+                                  if (showOnlyFavorites) "Show all trips"
```

Comment:
Nit: I wonder if we should use string resource instead of hardcoded strings as well for the contentDescription, since these strings are not really displayed anywhere in the app.

Reply:
I removed the hardcoded strings, I think this is a best practice for accessibility reasons (e.g. visually impaired users)